### PR TITLE
Releases feed parsing

### PR DESCRIPTION
**Pull request by csaunders**
This is totally a WIP but it should more correctly address the issue raised in #134.

I'm planning on getting the rest of this finished up during the Ruby Hack night tonight so the PR should be ready for review by the end of the day.

/cc @stephenminded @cshold

### CODE REVIEW

**Comment by csaunders** [comment]
Please review @girasquid @costford

File: lib/shopify_theme/releases.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def fetch!
+      response = HTTParty.get(FEED_URL)
+      raise "Could not retrieve feed from #{FEED_URL}" if response.code != 200
+      @feed = RSS::Parser.parse(response.body)
+    end
+
+    def all
+      @all ||= begin
+        versioned_releases.reduce({'master' => master, 'latest' => latest}) do |all, release|
```
Comment (girasquid) [inline lib/shopify_theme/releases.rb]: Why `reduce` over `group_by`?

File: spec/unit/releases_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        @releases.find('reccomended')
+      end
+    end
+
+    it "should provide a zip URL for a release" do
+      assert_equal "https://github.com/Shopify/Timber/archive/v2.0.2.zip", @releases.all['v2.0.2'].zip_url
+    end
+
+    it "should provide a zip URL for the latest release" do
+      assert_equal "https://github.com/Shopify/Timber/archive/v2.0.2.zip", @releases.all['latest'].zip_url
```
Comment (girasquid) [inline spec/unit/releases_spec.rb]: This test seems a bit brittle when the latest release changes - is that ok?

**Comment by girasquid** [comment]
Two questions, but looks good.

**Comment by stephenminded** [comment]
I think looking at the conversation in https://github.com/Shopify/shopify_theme/pull/134 you meant to cc @stevebosworth?

**Comment by csaunders** [comment]
@stephenminded yeah. Ooops. Too many steves.

Thanks for the review :dancer:

**Comment by stephenminded** [comment]
@csaunders oh yeah, forgot to give you a :+1: :smiley:

File: lib/shopify_theme/releases.rb (hunk 1 above)
Comment (csaunders) [inline lib/shopify_theme/releases.rb]: `master` and `latest` are elements I want to be included in the collection. I *could* just append the versioned releases and append them to an array containing the `master` and `latest` items as well.

I'm really just using a hash so I can easily grab the keys :P

Comment (girasquid) [inline lib/shopify_theme/releases.rb]: Fair enough!

File: spec/unit/releases_spec.rb (hunk 2 above)
Comment (csaunders) [inline spec/unit/releases_spec.rb]: I'm sort of OK with that. It's wrapped in VCR so it should only break if we update the cassette. Unless Github kills their atom feeds I don't see this breaking.

Worth the risk?

Comment (girasquid) [inline spec/unit/releases_spec.rb]: > It's wrapped in VCR so it should only break if we update the cassette.
> Worth the risk?

I think so, yeah - this works.

**Comment by costford** [comment]
Cool. LGTM!

**Pull request merged**